src/ui/RunEditor.tsx

```diff
@@ -850,33 +855,34 @@ export class RunEditor extends React.Component<Props, State> {
                             return null;
                         }
 
-                        const renderedVariables = [];
-
-                        const variables = gameInfo?.variables;
+                        const renderedVariableColumns = [];
+                        const renderedVariablesInExpandedRow = [];
+                        
                         if (variables != null) {
-                            for (const [keyId, valueId] of Object.entries(run.values)) {
-                                const variable = variables.data.find((v) => v.id === keyId);
-                                if (variable != null) {
+                            for (const variable of variables) {
+                                const valueId = run.values[variable.id];
+                                let valueName;
+                                if (valueId) {
                                     const value = Object.entries(variable.values.values).find(
                                         ([listValueId]) => listValueId === valueId,
                                     );
-                                    const valueName = map(value, (v) => v[1].label);
+                                    valueName = map(value, (v) => v[1].label);
                                     if (valueName != null) {
-                                        renderedVariables.push(
+                                        renderedVariablesInExpandedRow.push(
                                             <tr>
                                                 <td>{variable.name}:</td>
                                                 <td>{valueName}</td>
                                             </tr>,
                                         );
-                                    }
+                                    }  
```

Comment:
You have some trailing whitespace. You may want to enable trimming trailing whitespace in VS Code.

Reply:
We should probably add `tslint` to our CI to catch stuff like this.

Reply:
Yeah that would be good.

Reply:
https://github.com/LiveSplit/LiveSplitOne/pull/226